doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -181,3 +181,57 @@ post-launched VMs (VM1 and VM2).
+In this document we provide an overview about potential hardening techniques from user space applications perspective. Please consider those pointers when defining the security architecture and threat model for your application.
+
+.. note:: This not a definitive guide on the security technologies to be used or how to implement security. We provide general pointers not bounded to a specific OS or use-case.
+
+1. **Secure Feature Configurability**
+
+   - ACRN ensure a minimal control plane for the configuration of the memory regions boundaries and name handles. This is managed only by the SOS during the creation of the guest VM through the Device Model (DM).
+   - SOS Admin should refer to the usage guide for secure usage of the configuration flow.
+   - Create different permission/groups for ```admin``` role to segregate between other entity that might have access to SOS. For example, only admin permissions allow R/W/X on DM binary.
+   - Reference:  `Inter-VM communication Usage guide <https://projectacrn.github.io/latest/developer-guides/hld/ivshmem-hld.html>`_
+
+
+2. **Apply Access Control**
+
+   - Add restrictions based on behavior or subject/object rules around information flow and accesses.
+   - In SOS, consider the ```/dev/shm``` device node as a critical interface with special access requirement. Those requirements can be fulfilled using any of the existing opensource MAC technologies or even ACLs depending on the OS compatibility (Clear, Ubuntu, windows, etc..) and integration/management complexity.
+   - In UOS, the shared memory region can be accessed using ```mmap()``` of UIO device node. Please consider applying access control accordingly. Other complementary info can be found under:
+      * ```/sys/class/uio/uioX/device/resource2``` --> shared memory base address.
+      *  ```/sys/class/uio/uioX/device/config``` --> shared memory Size.
+   - For Linux based guest VMs, we recommend using the standard ```UIO``` and ```UIO_PCI_GENERIC``` drivers through the device node (Ex: ```/dev/uioX```)
+   - Reference: `AppArmour <https://wiki.ubuntuusers.de/AppArmor/>`_, `SELinux <https://selinuxproject.org/page/Main_Page>`_, `UIO driver-API <https://www.kernel.org/doc/html/v4.12/driver-api/uio-howto.html>`_
+
+
+3. **Crypto Support/Secure Applied Crypto**
+
+   - According to the application’s threat model and the defined assets that need to be shared securely, define the requirements for crypto algorithms. Those algorithms should enable operations like Authenticated Encryption/Decryption, Secure Key Exchange, True Random Number Generation, and Seed Extraction. In addition, consider the landscape of you attack surface and define the need for security engine (Ex: CSME) services.
+   - Don't implement your own crypto and use available Intel crypto libraries when applicable or other open source complaint libraries. (`Intel IPP <https://github.com/intel/ipp-crypto>`_, `TinyCrypt <https://01.org/tinycrypt>`_)
```

Comment:
I'd drop the "Intel" since this is an open source document, fix misspelling of "compliant", and rewrite this first sentence as:
```
Don't implement your own crypto functions. Use available compliant
crypto libraries as applicable, such as
`Intel IPP <https://github.com/intel/ipp-crypto>`_ or
`TinyCrypt <https://01.org/tinycrypt>`_.
```

Reply:
done